### PR TITLE
jenkins: remove macOS 10.x release machines

### PR DESCRIPTION
**Pull request by UlisesGascon**
### Main changes

This will remove macOS 10.x release machines in Jenkins pipelines.

### Context

On November 1, 2023 macOS 10.x won't be able to notarize the binaries.

Related: https://github.com/nodejs/build/issues/3385#issuecomment-1729281269
Related: https://github.com/nodejs/node/pull/50291

### CODE REVIEW

File: jenkins/scripts/VersionSelectorScript.groovy
Comment (richardlau) [inline jenkins/scripts/VersionSelectorScript.groovy:79]: We'll need to adjust `osx11-x64-release-tar` so that Node.js 18 runs on those instead of `osx1015-release-tar` otherwise we'll have no Node.js 18 macOS x64 tarballs.

Comment (UlisesGascon) [reply jenkins/scripts/VersionSelectorScript.groovy:79]: Not sure how to do that, can you guide me or provide an example? I was using an old PR as a reference for this change 🤔

Comment (targos) [reply jenkins/scripts/VersionSelectorScript.groovy:79]: If only one `release-tar` remains, you can remove the previous line, which currently excludes Node.js 18 from `osx11-x64-release-tar`.
Also, if there are no longer any `osx1015` configs, you can delete the first line.
In other words, if after this change all Node.js versions are supposed to be built on the same machine label, you can remove all osx-related lines from this file.

Comment (richardlau) [reply jenkins/scripts/VersionSelectorScript.groovy:79]: @UlisesGascon Quick overview of how this works:
- Job has all possible things it might run on added as configuration matrix selections.
- Job also has reference to this groovy script which places addition restrictions. The table at the top of this file lists where to exclude (i.e. do not run) configurations matching the regular expressions. If nothing excludes the configuration being matched against (i.e. no entry in this table) it is allowed.

As an aside, please don't merge/make any change to the set up until after tomorrow's Node.js 20 LTS transition release is done.

Comment (UlisesGascon) [reply jenkins/scripts/VersionSelectorScript.groovy]: Thanks for the explanations and the feedback @targos @richardlau 👍 

I made some changes in 146d9cf:
- I removed the references to `osx1015`
- I added support for Node18 in `osx11-x64-release-tar`
- I added `osx11-x64-release-pkg` so the pkg notarization can be done also (not sure if this is correct). As far I know we want to target pkg rather than tarball currently.

File: jenkins/scripts/VersionSelectorScript.groovy
Comment (richardlau) [inline jenkins/scripts/VersionSelectorScript.groovy:77]: `gte(18)` will mean that these will be excluded (i.e. not run) on all our currently supported versions of Node.js, which isn't what we want. 

Highlighting again @targos 's comment in https://github.com/nodejs/build/pull/3531#discussion_r1367893562 
> In other words, if after this change all Node.js versions are supposed to be built on the same machine label, you can remove all osx-related lines from this file.

Comment (UlisesGascon) [reply jenkins/scripts/VersionSelectorScript.groovy]: ok, I see it now much clear :)

I removed the references to macOS in [ee12477](https://github.com/nodejs/build/pull/3531/commits/ee124777935dcdf229a6972f4e1afd8dd009b381) because we will only have macOS11 machines available and all the Node.js versions (18-21) should use them. So there is no need to restrict any machine

**Comment by UlisesGascon** [comment]
My last question is.. if this remove the restrictions. How do I remove the machine tags for macos10.x? Is it done in the UI? 🤔

**Comment by targos** [comment]
You can either remove the label from the machine (in the "configure" tab) or remove the machine. When no machine has the label anymore, the label disappears.

**Comment by richardlau** [comment]
> You can either remove the label from the machine (in the "configure" tab) or remove the machine. When no machine has the label anymore, the label disappears.

Please remove the labels from the CI jobs first otherwise they'll try to run on machines with that label but none will exist leading to the job never completing.

**Comment by UlisesGascon** [comment]
> Please remove the labels from the CI jobs first otherwise they'll try to run on machines with that label but none will exist leading to the job never completing.

I added this as a requirement in https://github.com/nodejs/build/pull/3532

**Review by targos — APPROVED** [review]
LGTM but only merge **after** osx1015 is removed from release CI

**Comment by UlisesGascon** [comment]
> LGTM but only merge after osx1015 is removed from release CI

Can I do it already? I don't see any proposal release on going https://github.com/nodejs/node/pulls?q=is%3Apr+release+is%3Aopen+proposal

**Review by richardlau — APPROVED** [review]

**Comment by UlisesGascon** [comment]
Both Jenkins agents are removed, so I will merge the pr. 

Note: I was not able to directly remove/delete the tags as they are dynamic

**Comment by richardlau** [comment]
> Both Jenkins agents are removed, so I will merge the pr.
 
@UlisesGascon https://ci-release.nodejs.org/job/iojs+release/ is still showing 
image

**Comment by UlisesGascon** [comment]
This is strange. Even if you only select them, the `iojs+release` is skipping them (see: https://ci-release.nodejs.org/job/iojs+release-ulises-experimental/5/). Therefore, there is no error even if no machines are associated with this label. I did this test in `iojs+release-ulises-experimental` just few minutes ago.


Maybe the label is there due historical/previously use?

**Comment by richardlau** [comment]
> This is strange. Even if you only select them, the `iojs+release` is skipping them (see: https://ci-release.nodejs.org/job/iojs+release-ulises-experimental/5/). Therefore, there is no error even if no machines are associated with this label. I did this test in `iojs+release-ulises-experimental` just few minutes ago.
> 
> Maybe the label is there due historical/previously use?

@UlisesGascon https://ci-release.nodejs.org/job/iojs+release-ulises-experimental/5/ ran for `main` (e.g. Node.js 22) which means that it will skip macOS 10.15 due to https://github.com/nodejs/build/blob/a7cc7402837340d90b3a1c858da6af5ea8fe89ce/jenkins/scripts/VersionSelectorScript.groovy#L76. I've requested a test build on v18.x-staging, https://ci-release.nodejs.org/job/iojs+release-ulises-experimental/6/, and you can see from https://ci-release.nodejs.org/job/iojs+release-ulises-experimental/6/console that it is trying to run on the (now removed) osx1015 labels.
```console
19:15:50 Node.js major version: 18
19:15:50 Node.js version: 18.18.3
19:15:50 Running nodes
19:15:50 Triggering [iojs+release-ulises-experimental » osx11-x64-release-tar](https://ci-release.nodejs.org/job/iojs+release-ulises-experimental/nodes=osx11-x64-release-tar/)
19:15:50 Triggering [iojs+release-ulises-experimental » osx11-arm64-release-tar](https://ci-release.nodejs.org/job/iojs+release-ulises-experimental/nodes=osx11-arm64-release-tar/)
19:15:50 Triggering [iojs+release-ulises-experimental » osx1015-release-tar](https://ci-release.nodejs.org/job/iojs+release-ulises-experimental/nodes=osx1015-release-tar/)
19:15:50 Triggering [iojs+release-ulises-experimental » osx11-release-pkg](https://ci-release.nodejs.org/job/iojs+release-ulises-experimental/nodes=osx11-release-pkg/)
19:15:50 Triggering [iojs+release-ulises-experimental » osx1015-release-pkg](https://ci-release.nodejs.org/job/iojs+release-ulises-experimental/nodes=osx1015-release-pkg/)
```

Normally you would edit the job, e.g. https://ci-release.nodejs.org/job/iojs+release/configure, and remove the labels from "Configuration Matrix". I asked in https://github.com/nodejs/build/pull/3531#issuecomment-1779565236 to do that first before removing the machines since if you edit the jobs now it won't show the osx1015 labels as no machines are available with them so you cannot deselect them.

**Comment by targos** [comment]
IIRC you should just open the edit page, save it, and the config won't contain the labels anymore

**Comment by targos** [comment]
I did that. osx1015 isn't here anymore:
CleanShot 2023-10-27 at 08 11 54

**Pull request merged**
